### PR TITLE
Code of Conduct for BOINC on Github

### PR DESCRIPTION
**Pull request by Ageless93**
This is the Code of Conduct file that the work group worked on, and have approved of. To be approved of by the PMC. Committer comments welcome. 

When approved, see https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/ for how to add this one.

### CODE REVIEW

**Comment by Ageless93** [comment]
Additional information:
I have written this CoC, wrote an initial two versions of it, one with 'personal pointers' (you, we, us, etc.) and one without those. The work group liked the second one, but it needed some additions. So this is version three of that. 

If approved by all of you and the PMC, it can be saved into the root directory of Master and end up on https://github.com/BOINC/boinc/community

**Comment by brevilo** [comment]
👍

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Not everyone has the same level of expertise in language or coding, so be patient if an explanation needs to be given.
+
+•  Be welcoming
+
+Aim to welcome and support people of all backgrounds and identities, of any race, ethnicity, culture, national origin, color, immigration status, social and economic class, educational level, sex, sexual orientation, gender identity and expression, age, size, family status, political belief, religion, and mental and physical ability.
+
+•  Be respectful and friendly
+
+Respect each other and each other's point of view. Different people have different perspectives on issues. Frustration cannot turn into a personal attack. It is important to remember that a community where people feel uncomfortable or threatened is not a productive one. Also remember that with a world-wide community, English may not be the home language of everyone and since all communication happens via text, it's impossible to read what mood the other is in.
```
Comment (Uplinger) [inline CODE_OF_CONDUCT.md]: Can we change "home language" to read "first language" or "native language"?

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Nobody is expected to be perfect in this community. Asking questions early avoids many problems later, so questions are encouraged. Those who answer should be responsive and helpful.
+
+•  Value decisiveness, clarity and consensus
+
+Disagreements, social and technical, are normal, but let's not allow them to persist and fester, leaving others uncertain of the agreed direction. The strength of the BOINC community comes from its diversity, people from a wide range of backgrounds. Being unable to understand why someone holds a viewpoint doesn't mean that that viewpoint is wrong.
+
+•  Resolve issues
+
+Don't forget that it is human to err and that blaming each other isn't productive. Instead, focus on helping to resolve issues and learning from mistakes. Participants to the project should try to resolve disagreements constructively. When it is impossible to reach an agreement, follow the [process outlined in the Governance document](https://github.com/BOINC/boinc-policy/blob/master/Governance.md#5-decision-processes)to come to a solution.
```
Comment (Uplinger) [inline CODE_OF_CONDUCT.md]: Should the start read like this instead?  Don't forget that to err is human and...

https://en.wiktionary.org/wiki/to_err_is_human

**Comment by Uplinger** [comment]
My comments are minor as you could probably tell.  As a whole, I like the document and would approve it as is.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (Ageless93) [reply CODE_OF_CONDUCT.md]: Okay, I can live with that. I'll change it to 'native language'.

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (Ageless93) [reply CODE_OF_CONDUCT.md]: If you don't mind, I keep it like this. It was like that in the original text I took parts of the CoC from at [the ToDoGroup](http://todogroup.org/opencodeofconduct/) (in the Try to understand why we disagree section).

Comment (Uplinger) [reply CODE_OF_CONDUCT.md]: I don't mind at all.  I could say this can be closed.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (Uplinger) [reply CODE_OF_CONDUCT.md]: Thanks, I see the commit below.

**Comment by ballen4705** [comment]
Ordering of bullets could be improved. Put "Be patient" before or after "Ask for help when unsure"

"since all communication happens by text" -> "since most communication happens by text"

"BOINC is a complex whole made of many parts, it is the sum of many dreams" is a run-on sentence. Change comma to colon.

"Working together between teams" -> Cooperation between teams

Change semicolon after last sentence to period.

"Disagreements, social and technical, are normal..." -> Social and technical disagreements should not be left to fester, leaving others uncertain of the direction.  The strength of the BOINC community comes from its diversity.  (I suggest dropping the last sentence, which I don't always agree with.)

Participants to the project -> Participants in the project

We further adhere -> We also adhere

**Comment by Ageless93** [comment]
> "since all communication happens by text" -> "since most communication happens by text"

This is the CoC for BOINC Github. Here, as far as I know, all communications happens by text. How are people communicating in other ways to everyone else? 

> "BOINC is a complex whole made of many parts, it is the sum of many dreams" is a run-on sentence. Change comma to colon.

It would be a run-on sentence without the comma. I don't see how a colon would increase legibility. If anything, I'd rather go for dropping ", it is the sum of many dreams." - mostly because I don't know how that got in there. 

>  I suggest dropping the last sentence, which I don't always agree with. ("Being unable to understand why someone holds a viewpoint doesn't mean that that viewpoint is wrong.")

But at times you do agree with it then. Reason enough to leave it in, aside from just one person suggesting for it to be removed, which isn't enough for me.

**Comment by TheAspens** [comment]
The BOINC PMC has voted 6-0 (2 people did not vote) to approve this.  As a result I am merging it.

**Comment by brevilo** [comment]
There's a git regression: this file uses Windows line breaks but it wasn't added to `.gitattributes` to say so, such that a pull of it on a Linux system causes an ever-dirty working tree:

```
warning: CRLF will be replaced by LF in CODE_OF_CONDUCT.md.
The file will have its original line endings in your working directory.
```

Please either add it to `.gitattributes` or use Linux line breaks that even Windows' Notepad understands nowadays 😏 

Thanks

**Comment by Ageless93** [comment]
I wrote the original in Notepad++, which does Linux linebreaks by default.
I've only later edited versions directly in the Github editor, so perhaps
that that did it.

How do I go about adding it to .gitattributes? I'll do so then later
tonight when I am home.

On Friday, May 18, 2018, Oliver Bock <notifications@github.com> wrote:
> There's a git regression: this file uses Windows line breaks but it
wasn't added to .gitattributes to say so, such that a pull of it on a Linux
system causes an ever-dirty working tree:
>
> warning: CRLF will be replaced by LF in CODE_OF_CONDUCT.md.
> The file will have its original line endings in your working directory.
>
> Please either add it to .gitattributes or use Linux line breaks that even
Windows' Notepad understands nowadays 😏
>
> Thanks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci5.googleusercontent.com/proxy/4O0eJa10LN21oDo0BGQn1HCBJhGoHW9JS6ao0B0BNpjmM_NkchaPGP8gWUawVyJz_3mHc_H1uhqKaQIj4LR9wpJ9x_o3s_0zmgpuoI53LREFpYxEJPWITMV2AXzaC6y-x95A-0gWsVO4ft2uWyV88rmRy9GZQA=s0-d-e1-ft#https://github.com/notifications/beacon/AKXbU-LXcvJDwuKioR_SnRpivaGRklxyks5tzoGggaJpZM4S9mZh.gif
>

-- 
Jord.

**Comment by Ageless93** [comment]
I didn't quote. I sent these answers by email from my phone via the Gmail
app.

On Friday, May 18, 2018, Oliver Bock <notifications@github.com> wrote:
> Please see my posting above.
> Cheers
>
> PS: your quoting is broken
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci5.googleusercontent.com/proxy/Z5RPaAHXpBNnixYLxVVG2yBXb0jkVfCK4j0p8pA2br4N-UxeAAw67iJbf7axUrnZEkvwzuPAfl9Ei7QBsQUE11Th-Gl-0ceEPxLqpXKi2g6bZVWLYTFX7ivoMiVpYmW1eZ4WIfq7b3A0AdaDz0hlNHmPY1Egdg=s0-d-e1-ft#https://github.com/notifications/beacon/AKXbU86o3UTvnG-1AZEjQZcr0h55qhI3ks5tzr14gaJpZM4S9mZh.gif
>

-- 
Jord.

**Comment by brevilo** [comment]
Yep, I figured that. Then Gmail did 😛 

Have a nice weekend

**Pull request merged**